backend/src/layer/data/flickr_api/flickr_api_handler.rs

```diff
@@ -37,17 +39,45 @@ impl FlickrApiHandler {
     // URL TYPE 1.2: https://www.flickr.com/photos/198319418@N06/53077317043 <- remove last '/'
     // URL TYPE 2: https://flic.kr/p/2oRguN3
     // Both cases: Split with '/' and get last member (= photo_id).
-    fn determine_photo_id(mut url: &str) -> Result<&str> {
-        if !SHORT_URL_REGEX.is_match(url) && !LONG_URL_REGEX.is_match(url) {
-            return Err(ImageHosterError::FormatNotFound(format!(
+    fn determine_photo_id(url: &str) -> Result<String> {
+        if LONG_URL_REGEX.is_match(url) || SHORT_URL_REGEX.is_match(url) {
+            let id = url
+                .trim_end_matches('/')
+                .split('/')
+                .last()
+                .map(String::from)
+                .ok_or_else(|| {
+                    ImageHosterError::FormatNotFound(format!(
+                        "this url format is not supported: '{url}'"
+                    ))
+                })?;
+            if SHORT_URL_REGEX.is_match(url) {
+                Self::decode(&id)
+            } else {
+                Ok(id)
+            }
```

Comment:
you unnseccsearily call regex.is_match two times for SHORT_URL_REGEX.
You can just split these into two seperate cases.
The you also could easily let the regex do the extracting for you and you dont have to do the whole tim_end, split part.